packages/bits/src/lib/time-picker/time-picker.component.ts

```diff
@@ -142,14 +152,34 @@ export class TimePickerComponent implements OnInit, OnDestroy, OnChanges, AfterV
         this.overlay.clickOutside
             .pipe(takeUntil(this.onDestroy$))
             .subscribe(_ => this.overlay.hide());
-        
+
         this.initPopupUtilities();
+        this.keyboardService.initService(this.popup.menuItems, this.popup, this.overlay, this.menuTrigger.nativeElement);
+        this.cdr.detectChanges();
     }
 
     onChange(value: any) {}
 
     onTouched() {}
 
+    @HostListener("keydown", ["$event"])
+    public onKeyDown(event: KeyboardEvent): void {
+        this.keyboardService.onKeyDown(event);
+    }
+
+    @HostListener("focusout")
+    onFocusOut(): void {
+        if (!this.overlay.showing || !document.activeElement) {
+            return;
+        }
+
+        setTimeout(() => {
```

Comment:
Why do we need `setTiemout` here? An explicit comment would be great to have for the future reference